src/simsoptpp/surface.cpp

```diff
@@ -138,6 +138,44 @@ void Surface<Array>::extend_via_normal(double scale) {
     this->least_squares_fit(target_values);
 }
 
+template<class Array>
+void Surface<Array>::extend_via_projected_normal(const vector<double> phi, double scale) {
```

Comment:
Agreed, it is mainly in the c++ code because it is extremely similar to the extend_via_normal c++ function so I will hold off on changing this to python for now.